### PR TITLE
[feature](nereids) add rewrite rule PushCountIntoUnionAll

### PR DESCRIPTION
**Pull request by feiniaofeiafei**
add transform:
```java
/**
 * LogicalAggregate  (groupByExpr=[c1#13], outputExpr=[c1#13, count(c1#13) AS `count(c1)`#15])
 *  +--LogicalUnion (outputs=[c1#13], regularChildrenOutputs=[[c1#9], [a#4], [a#7]])
 *    |--child1 (output = [[c1#9]])
 *    |--child2 (output = [[a#4]])
 *    +--child3 (output = [[a#7]])
 * transform to:
 * LogicalAggregate (groupByExpr=[c1#13], outputExpr=[c1#13, sum0(count(c1)#19) AS `count(c1)`#15])
 *  +--LogicalUnion (outputs=[c1#13, count(c1)#19], regularChildrenOutputs=[[c1#9, count(c1)#16],
 *   [a#4, count(a)#17], [a#7, count(a)#18]])
 *    |--LogicalAggregate (groupByExpr=[c1#9], outputExpr=[c1#9, count(c1#9) AS `count(c1)`#16])
 *    |  +--child1
 *    |--LogicalAggregate (groupByExpr=[a#4], outputExpr=[a#4, count(a#4) AS `count(a)`#17])
 *    |  +--child2
 *    +--LogicalAggregate (groupByExpr=[a#7], outputExpr=[a#7, count(a#7) AS `count(a)`#18]]
 *      +--child3
 */
```

### CODE REVIEW

**Comment by feiniaofeiafei** [comment]
run build all

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/CheckAfterRewrite.java [hunk 1]
```diff
@@ -106,7 +106,7 @@ private void checkAllSlotReferenceFromChildren(Plan plan) {
         if (notFromChildren.isEmpty()) {
             return;
         }
-
+        System.out.println(plan.treeString());
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/analysis/CheckAfterRewrite.java:109]: remove this line

**Comment by feiniaofeiafei** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushCountIntoUnionAll.java [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+        // create the new LogicalUnion
+        LogicalSetOperation newLogicalUnion = logicalUnion.withChildrenAndTheirOutputs(newChildren.build(),
+                childrenOutputs.build());
+        List<NamedExpression> newLogicalUnionOutputs = Lists.newArrayList();
+        for (NamedExpression ce : upperOutputExpressions) {
+            if (ce instanceof Alias) {
+                newLogicalUnionOutputs.add(new SlotReference(ce.getName(), ce.getDataType(), ce.nullable()));
+            } else if (ce instanceof SlotReference) {
+                newLogicalUnionOutputs.add(ce);
+            }
```
Comment (924060929) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/rewrite/PushCountIntoUnionAll.java:108]: add else condition to return origin LogicalUnion, means you can not process this cases

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run p0

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Review by morrySnow — COMMENTED** [review]
should add enough ut and regression test as we discussion

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Review by 924060929 — DISMISSED** [review]

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run cloud_p0

**Comment by feiniaofeiafei** [comment]
run buildall

**Comment by feiniaofeiafei** [comment]
run cloud_p0

**Review by 924060929 — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**